modules/alloy/src/main/ts/ephox/alloy/behaviour/docking/DockingApis.ts

```diff
@@ -85,11 +85,14 @@ const refreshInternal = (component: AlloyComponent, config: DockingConfig, state
   // If docked then check if we need to hide/show the component
   const isDocked = state.isDocked();
 
-  // Checking isDocked works for only, toolbar_location: top mode, as you scroll, it would be undocked, docked then hidden
-  // But for toobar_location, it would be from docked, undocked then hidden
-  if (isDocked || (!isDocked && Arr.contains(state.getModes(), 'bottom'))) {
+  viewport.optScrollEnv.fold(() => {
```

Comment:
Not sure if this is the appropriate way? `optScrollEnv` would only be set when `ui_mode: 'split'`.

Reply:
```suggestion
 if ((ui_mode === 'split' && viewport.optScrollEnv.isSome()) || isDocked) {
   updateVisibility(component, config, state, viewport);
 }
```
I'm happy with either way. This is assuming `optScrollEnv` is an option and you only want to `updateVisibility` when `isDocked` or within a scrolling environment.

Reply:
Ah yeah, `isSome()` works as well. Not sure if we want `ui_mode` option in Alloy?

Reply:
Hmm, if this is only relevant in a scrollable container, and doesn't have to do with the sink, perhaps it can be included without depending on the option?

Reply:
I think the main intention is to reduce unnecessary calls. In `ui_mode: combined` mode, the docking only happens at the top/bottom, so updating the visibility occurs only when it's docked, e.g. for docking at the top:
1. Toolbar location top, Scrolling to the bottom of the screen, the toolbar will be docked (fixed position). Scrolling further down the page, it's not partly visible; it will be hidden (changing from docked mode [fixed], needs to be hidden cause docking at the top has a fixed position). Scrolling from bottom to top, the editor and toolbar won't be in the viewport. Because it is statically positioned (undocked mode), so the toolbar won't be visible even if it's not hidden anyway. 
But I think it makes sense to remove the `isDocked` checking and update visibility on scroll.